### ISSUE TITLE
WCAG 2.1: Buttons' inner background color has insufficient contrast against its adjacent background

### ISSUE THREAD

**Issue by iamrafan** [microsoft/accessibility-insights-web#762]
## Describe the bug

Related to [WCAG SC 1.4.11: Non-text Contrast](https://www.w3.org/WAI/WCAG21/Understanding/non-text-contrast)

>For active controls such as buttons and form fields: any visual information provided that is necessary for a user to identify that a control is present and how to operate it must have a minimum 3:1 contrast ratio with the adjacent colors

**To Reproduce**

1. Run assessment on any page
2. Open details view and start an assessment
3. Open the target page
4. Navigate to a new URL in the same tab
5. Run assessment once again
6. A dialog 'Assessment in progress' shows up in details view window
7. Observe both the ```Continue previous``` and ```Start new``` buttons have a background color that isn't having enough contrast with the adjacent background

![image](https://user-images.githubusercontent.com/22068434/59233438-4a0cf400-8b9d-11e9-86e7-aa7a79e1621b.png)

## Expected behavior

The contrast ratio between the buttons' background and the adjacent background should be at least 3:1

## Screenshots

![image](https://user-images.githubusercontent.com/22068434/59233077-d6b6b280-8b9b-11e9-84c1-37ac58498345.png)

## Extension

- Accessibility Insights for Web - prod
- Browser Version: Chrome 75.0.3770.80
- Target Page: Any page

## Did you search for similar existing issues?
Yes - #761

**Comment by cheade** [comment]
I'd like to propose this color scheme for our secondary buttons in order to be in accordance with WCAG 2.1:
![image](https://user-images.githubusercontent.com/40274682/59383327-8747c200-8d14-11e9-83f8-5f1e3d45101b.png)

**Button styling**
background-color: #FFFFFF;
border-color: rgba(0, 0, 0, 0.42) /* we'll need to add new shade, --neutral-alpha-42
border-radius: 2px;

**Text**
color: $primary-text

The border, being alpha, will pull in 3:03:1 on white backgrounds, and 3.01:1 on our light grey (#FAFAFA) backgrounds

**Comment by ferBonnin** [comment]
@cheade,  thanks! but right now we are using black border to indicate element has focus. How would we show element has focus?

**Comment by cheade** [comment]
@ferBonnin We could tackle it the way new Fluent Fabric does! The buttons are the same except for modified colors:
![image](https://user-images.githubusercontent.com/40274682/59394287-e963ef80-8d33-11e9-9746-8d9d7cb0b261.png)

The CSS for these buttons can be [pulled from here](https://www.figma.com/file/Wj4Ggf6GGQBQkiDIaHfXRX2B/Accessibility-Insights-Style-guide?node-id=13%3A53)

**Comment by ferBonnin** [comment]
thanks Chelsea!
 @iamrafan thoughts on the approach?

**Comment by iamrafan** [comment]
@ferBonnin Yes, I think this approach would satisfy the requirement for 1.4.11 provided the button's border in all the cases is having enough contrast against the button's background and adjacent background.

Is the border in focus state having enough contrast? 
![image](https://user-images.githubusercontent.com/22068434/59454578-42c92e80-8dc7-11e9-9c90-1dcc86f313bd.png)

**Comment by ferBonnin** [comment]
from the figma file, the border color is #666666 should be fine.

**Comment by cheade** [comment]
On the topic of focus styles, since we like the dotted line focus style around the gear icon, can we have our buttons match? (it's the same styling as the gear icon's focus)

![image](https://user-images.githubusercontent.com/40274682/59542097-1a216180-8eb9-11e9-8c6c-259b7ffc0096.png)

**Comment by ferBonnin** [comment]
i think we only have the dotted line in high contrast mode, but will check when triagging if possible.

**Comment by cheade** [comment]
The dotted focus line appears to be only on the drop-down in both high contrast and normal mode. (it doesn't have a dotted line around buttons or any thing else) Everything else has a solid line, but it seems like a solid line might be harder to follow. I think the dotted line brings a nice visual to the focus on all the elements :).

![image](https://user-images.githubusercontent.com/40274682/59542405-7cc72d00-8eba-11e9-850b-be5329806b23.png)

**Comment by ferBonnin** [comment]
per conversation with Peter, convert this into a feature including the high contrast stuff and investigating what officeFabric is doing. @ferBonnin

**Comment by ferBonnin** [comment]
@cheade having two borders for buttons that receive focus is complex. Could we simply have the outer border and have it being dotted when it receives focus.

**Comment by cheade** [comment]
@ferBonnin we could for the standard/secondary button, but for the primary button there might be trouble. Black is hard to see against the blue and white can’t be see against the normal background of the page.

I know that updated fabric has the focus indicator inside the button (but it’s solid), are we able to leverage that?

**Comment by ferBonnin** [comment]
will need to investigate if we can leverage this. will create a feature.

**Comment by ferBonnin** [comment]
@iamrafan this looks to be addressed by OF V7 update. can you confirm?

**Comment by iamrafan** [comment]
@ferBonnin Yes, this issue has been fixed
